### DISCUSSION TITLE
Question: What would be the downside of having two msw server instances running at the same time?

### DISCUSSION THREAD

**Discussion by brianzhou13** [mswjs/msw#821] [Q&A] [answered]
## Environment

n/a

## Request handlers

So we have individual test files that'll spin up their own msw servers with `setupServer`:

```js
// test-specific-file.test.ts

const server = setupServer(...handlers);

describe('a test', () => {
    beforeAll(() => server.listen());
    afterEach(() => server.resetHandlers());
    afterAll(() => server.close());
    test('api requests', () => {
      // ...
    })
})
```

Then in our jest setup, we spin up an additional server before each test run to mock out any accidentally non-mocked routes:

```
// setupFilesAfterEnv.js

beforeAll(() => globalServer.listen());
afterEach(() => globalServer.resetHandlers());
afterAll(() => globalServer.close());
```

## Actual request

Mostly with `Axios`, but probably some sprinkles of `fetch`

## Current behavior

The tests work correctly, but the downside of doing so isn't clear to me, and that's what I'm hoping to answer through this GH issue.

## Expected behavior

That the tests run properly.

## Screenshots

n/a

**Comment by kettanaito** [comment, marked as answer]
Hey, @brianzhou13.

There's little reason to have multiple instances of `setupServer` calls. If anything, that will patch the request-issuing modules multiple times (which is how request interception works in Node.js), which may lead to all sorts of unexpected behavior. 

Instead, create a _single_ `setupServer` integration point in your global test setup and modify it per test suite, if necessary. That way you get the same control over your API with a single setup. 

```js
import { setupServer } from 'msw'

export const server = setupServer(/* global handlers, if any */)
```

```js
// setupFilesAfterEnv.js
import { server } from './server'

beforeAll(() => server.listen())
afterEach(() => server.resetHandlers())
afterAll(() => server.close())
```

```js
// some.test.js
import { msw } from 'msw'
import { server } from './server'

beforeEach(() => {
  server.use(rest.get('/', resolver))
})
```

Note that if you decide to use `.resetHandlers()` in the global `afterEach` hook, then applying a test suite-wide handlers would be better via the `beforeEach` hook, since all runtime handlers you add via `.use()` get removed as the result of the `afterEach` global hook.

**Reply by mihilranathunga** [reply]
how does this affect parellel test runs? does server.use from one test suit affect another since they have the same server setup?
